### DISCUSSION TITLE
RuCore / Профили типов данных / Core_Address

### DISCUSSION THREAD

**Discussion by eugenekogan-spb** [fhir-ru/core#237] [Минимальный набор артефактов]
```
Profile: Core_Address
Id:      core-address
Parent:       Address
Title: "Адрес (Ru Core)"
Description: "Адрес"

* use ^short = "Целевое назначение адреса: home | work | temp | old | billing"

* type ^short = "Тип адреса: postal | physical | both"

* text ^short = "Текстовое представление адреса"

* line ^short = "Улица"

* city ^short = "Город"

* state ^short = "Регион"

* postalCode ^short = "Почтовый код"

* country ^short = "Страна"

* period ^short = "Период времени, когда адрес был/является действительным"

* extension contains $Ядро_Расширение_КодыФИАС named fias 0..1 MS


Instance:   core-address-instance-full
InstanceOf: Core_Address
Usage: #inline

* use = #home

* type = #postal

* text = "Россия, г. Санкт-Петербург, Софийская ул., д.32-1-987"

* line = "Софийская 32-1-987"

* city = "Санкт-Петербург"

* state = "Санкт-Петербург"

* postalCode = "192236"

* country = "Россия"

* extension[fias]
  * extension[aoguid]
    * valueIdentifier
      * value = "7b1996be-c0fe-4797-bedc-f503e31a83fd"
      * system = "urn:hl7-ru:fias:aoguid"
  * extension[houseguid]
    * valueIdentifier
      * value = "7b1996be-c0fe-4797-bedc-f503e31a83fd"
      * system = "urn:hl7-ru:fias:houseguid"
```

Методические рекомендации по профилированию ресурсов и типов данных: https://github.com/fhir-ru/core/discussions/248

**Comment by eugenekogan-spb** [comment]
Поправлено для отражения комплексного расширения - [238](https://github.com/fhir-ru/core/discussions/238), на основании решений описанных в [242](https://github.com/fhir-ru/core/discussions/242#discussioncomment-7457586).

**Reply by Vrodvi** [reply]
Замечание закрыто.

**Comment by Vrodvi** [comment]
# Замечание 1.

В соответствии с базовой спецификацией (см. https://www.hl7.org/fhir/datatypes.html#identifier):
префикс "urn:uuid:" должен использоваться для глобально уникальных идентификаторов UUID, но тогда и система идентификации должна быть обозначена как "urn:ietf:rfc:3986"
![image](https://github.com/fhir-ru/core/assets/92928137/ed953236-d1f4-400a-a002-8ee7364028c3)

В нашем же случае (см. структуру тут https://github.com/fhir-ru/core/discussions/238) - установлены две локальные системы идентификации (urn:hl7-ru:fias:aoguid и urn:hl7-ru:fias:houseguid), поэтому префикс не нужен. Считаем что идентификаторы, выглядящие как UUID - это просто совпадение. Т.е. указываем идентификатор как есть - просто буквенно-циферные последовательности через тире.

Рекомендуется убрать префиксы "urn:uuid:" из примера.

**Reply by eugenekogan-spb** [reply]
Префиксы из примера убраны.

**Reply by Vrodvi** [reply]
Замечание "1." закрыто.

**Comment by Vrodvi** [comment]
# Замечание 2. 

Необходима страница с описанием регламента / лучших практик по формирования данного типа артефактов - "Профиль типа данных"

Была договорённость, что результатом работы по каждому типу артефакта будет, в том числе, и специальная страничка/документ с описанием особенностей/рекомендаций по формированию данного типа артефактов.

**Reply by eugenekogan-spb** [reply]
Дана ссылка на страницу с рекомендациями.

**Reply by Vrodvi** [reply]
Замечание "2." закрыто.

**Comment by Vrodvi** [comment]
# Замечание 3.

Лишние ограничения в профиле.

Записи вида `* use 0..1` выглядят как наложение ограничений на кардинальность атрибута, хотя по факту никаких ограничений не накладывается.

Если есть необходимость обозначить, что данный атрибут должны уметь получать, разбирать и интерпретировать все системы, работающие с данным Ru-Core-профилем, то возможно более подходящим будет использовать вместо указания кардинальности указание отметки Must Support (обсуждали её вот тут - https://github.com/fhir-ru/core/discussions/38).

Если указано было "просто так", то я бы рекомендовал это убрать.

**Reply by eugenekogan-spb** [reply]
Указано "просто так". Убрано.

**Reply by eugenekogan-spb** [reply]
В дискуссию https://github.com/fhir-ru/core/discussions/38 добавил комментарий о фактическом использовании флага MustSupport в прикладном руководстве - по СЭМД ПЛИ.

**Reply by Vrodvi** [reply]
Замечание "3." закрыто.

**Comment by Vrodvi** [comment]
# Замечание 4. 

Поскольку FSH является специализированным языком разметки для использования людьми, а не машинами, то форматирование описаний желательно оформлять по каким-то общим стилям. 

К примеру, почему в верхнем примере все атрибуты разделены пустой строкой, а в нижнем примере пустая строка имеется только между двумя частями расширения? Какой смысл закладывается в подобного рода форматирование?

Когда описание сформировано не унифицировано, то глаз зацепляется за подобные расхождения и это вынуждает начать поиск дополнительного смысла там, где он не закладывался.

**Reply by eugenekogan-spb** [reply]
Согласен. Приведено к единому виду. В предложенном варианте: атрибуты верхнего уровня разделены пустой строкой, атрибуты следующего уровня описаны "лесенкой".

**Reply by Vrodvi** [reply]
Замечание "4." закрыто.

**Comment by Vrodvi** [comment]
# Замечание 5.

Для экземпляров в атрибуте Instance необходимо использовать соглашения как для идентификаторов (id), а не для наименований:
`Instance: CoreAddressInstanceFull`
(подробнее - см. https://github.com/fhir-ru/core/discussions/242#discussioncomment-7648882)

**Reply by eugenekogan-spb** [reply]
Исправлено.

**Reply by Vrodvi** [reply]
Замечание "5." закрыто.
